src/data/fin.lean

```diff
@@ -357,6 +357,12 @@ def cast (eq : n = m) : fin n ≃o fin m :=
 
 @[simp] lemma coe_cast (h : n = m) (i : fin n) : (cast h i : ℕ) = i := rfl
 
+@[simp] lemma cast_trans {k : ℕ} (h : n = m) (h' : m = k) {i : fin n} :
+  cast h' (cast h i) = cast (eq.trans h h') i := rfl
+
+@[simp] lemma cast_id {h : n = n} {i : fin n} : cast h i = i :=
```

Comment:
Should this be `cast_refl` instead of `id`, since h is rfl?  Does the lemma still apply if you remove the hypothesis and use `rfl` directly?

Reply:
i.e.:
```suggestion
@[simp] lemma cast_refl {i : fin n} : cast rfl i = i :=
```